rpc-rs/src/otel.rs

```diff
@@ -36,11 +37,14 @@ impl<'a> OtelHeaderExtractor<'a> {
 
 impl<'a> Extractor for OtelHeaderExtractor<'a> {
     fn get(&self, key: &str) -> Option<&str> {
-        self.inner.get(key).and_then(|s| s.to_str().ok())
+        self.inner.get(key).and_then(|s| s.iter().next().map(|v| v.as_str()))
     }
 
     fn keys(&self) -> Vec<&str> {
-        self.inner.keys().map(|s| s.as_str()).collect()
+        self.inner
+            .iter()
+            .map(|(k, _)| std::str::from_utf8(k.as_ref()).unwrap_or(""))
```

Comment:
This could probably be fine with a `unwrap()` call since the underlying `HeaderName` is wrapping a `String`. What do you think?